src/Compatibility/Core/src/Android/Renderers/AndroidGIFImageParser.cs

```diff
@@ -218,18 +218,18 @@ public static async Task<IFormsAnimationDrawable> LoadImageAnimationAsync(FileIm
 
 		public static async Task<IFormsAnimationDrawable> LoadImageAnimationAsync(UriImageSource imagesource, Context context, CancellationToken cancelationToken = default(CancellationToken))
 		{
-			Uri uri = imagesource?.Uri;
 			FormsAnimationDrawable animation = null;
 
-			if (uri != null)
+			if (imagesource is IStreamImageSource streamImageSource)
 			{
-				var options = new BitmapFactory.Options
+				using var stream = await streamImageSource.GetStreamAsync(cancelationToken).ConfigureAwait(false);
+				if (stream != null)
 				{
-					InJustDecodeBounds = true
-				};
+					var options = new BitmapFactory.Options
+					{
+						InJustDecodeBounds = true
+					};
 
-				using (Stream stream = await imagesource.GetStreamAsync(cancelationToken).ConfigureAwait(false))
-				{
 					using (var decoder = new AndroidGIFImageParser(context, options.InDensity, options.InTargetDensity))
```

Comment:
I wonder if we should just delete this whole thing from Compatibility and use glide instead compatibility also. The `AndroidGIFImageParser` has always been problematic. Don't have to cross this bridge now but maybe....

Reply:
Yeah. This is part of the plan. I have an issue to fix this: #867